### PR TITLE
Feat text format

### PR DESCRIPTION
**Pull request by artem30801**
## What type of pull request is this?

- [X] Non-breaking code change
- [ ] Breaking code change
- [ ] Documentation change/addition

## Description
Added utils for text formatting for discord messages and utils for ansi formatting

Examples of some usages:
```py
from dis_snek.client.utils import formatting as text

msg = text.bold("I'm bold")
msg = f"Bold name: {text.bold(ctx.author.display_name)}"
msg = text.italic(text.bold("And italic"))
msg = text.link_in_embed("google", "https://google.com/")
msg = text.code_block("a = 25  # python code", language="py")

# Manual ansi styling
msg = text.ansi_block(f"{text.ansi_format(color=text.colors.RED)} Text in red{text.ansi_escape} Back in black!")

# Wrapper function ansi styling
msg = text.ansi_block(
    f"{text.ansi_styled('Hello there', text.styles.UNDERLINE, text.colors.BLUE, text.bg_colors.INDIGO)}"
    f" Then Normal text"
    f"{text.ansi_styled('Hello there too', text.styles.BOLD, text.colors.RED, text.bg_colors.SOME_VERY_DARK_BLUE)}"
)
```

## Changes

- Added utils for classical text formatting eg bold italic, spoilers, code blocks etc
- Added utils for formatting/coloring text in ansi code blocks

## Checklist

- [X] I've formatted my code with [Black](https://black.readthedocs.io/en/stable/)
- [X] I've ensured my code works on `Python 3.10.x`
- [X] I've tested my code

### CODE REVIEW

File: dis_snek/client/utils/formatting.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+
+def link_in_embed(text: str, url: str) -> str:
+    """Makes a clickable link inside embed"""
+    return f"[{text}]({url})"
+
+
+def quote_line(line: str) -> str:
+    """Formats a text line for discord message as quote"""
+    return f">{line}"
```
Comment (AlbertUnruh) [inline dis_snek/client/utils/formatting.py]: As far I know Discord wants a SPACE between `>` and `QUOTED TEXT`

Comment (artem30801) [reply dis_snek/client/utils/formatting.py]: Good catch

**Review by Kigstn — APPROVED** [review]
Love it

**Comment by AlbertUnruh** [comment]
The only thing I'm missing are `__all__` and the entry (`from .formatting import *`) in `__init__.py` ^^
But nice feature 👍

**Review by LordOfPolls — APPROVED** [review]
Once this has an `__all__` entry, it can be merged either by myself, or another org member

**Pull request merged**